cmd/cstor-volume-mgmt/volume/volume.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+)
+
+// VolumeOperator is the name of the tool that makes volume-related operations.
+const (
+	VolumeOperator  = "iscsi"
+	IstgtConfPath   = "/usr/local/etc/istgt/istgt.conf"
+	IstgtStatusCmd  = "STATUS"
+	IstgtRefreshCmd = "REFRESH"
+)
+
+//FileOperatorVar is
+var FileOperatorVar util.FileOperator
+
+//UnixSockVar is
+var UnixSockVar util.UnixSock
+
+// CreateVolume creates a new cStor volume istgt config.
+func CreateVolume(cStorVolume *apis.CStorVolume) error {
+	// create conf file
+	text := CreateIstgtConf(cStorVolume)
+	err := FileOperatorVar.Write(IstgtConfPath, text, 0644)
```

Comment:
Is it possible to get two get requests for same volume? If yes then this should be thread safe.

Reply:
As we are watching only for a single volume CR, we won't get multiple parallel requests.